compiler/rustc_typeck/src/collect.rs

```diff
@@ -94,7 +94,12 @@ pub fn provide(providers: &mut Providers) {
 ///////////////////////////////////////////////////////////////////////////
 
 /// Context specific to some particular item. This is what implements
-/// `AstConv`. It has information about the predicates that are defined
+/// `AstConv`.
+///
+/// `ItemCtxt` is primarily used to type-check item signatures, in contrast to [`FnCtxt`],
+/// which is used to type-check function bodies.
```

Comment:
Function bodies is probably the best way to put it here, but maybe also note that it's used to typecheck things like closures and consts too. Things that have bodies where expressions and statements live, basically.